skrifa/src/color/traversal_tests.rs

```diff
@@ -0,0 +1,341 @@
+};
+
+#[derive(Default)]
+struct JsonDumpColorPainter {
+    paint_dump: PaintDump,
+}
+
+#[derive(Serialize, Deserialize, Default, PartialEq)]
+struct PaintDump {
+    glyph_id: u16,
+    ops: Vec<PaintOps>,
+}
+
+impl From<Brush<'_>> for BrushParams {
+    fn from(brush: Brush) -> Self {
+        match brush {
+            Brush::Solid {
+                palette_index,
+                alpha,
+            } => BrushParams::Solid {
+                palette_index,
+                alpha,
+            },
+            Brush::LinearGradient {
+                p0,
+                p1,
+                color_stops,
+                extend,
+            } => BrushParams::LinearGradient {
+                p0,
+                p1,
+                color_stops: color_stops.to_vec(),
+                extend,
+            },
+            Brush::RadialGradient {
+                c0,
+                r0,
+                c1,
+                r1,
+                color_stops,
+                extend,
+            } => BrushParams::RadialGradient {
+                c0,
+                r0,
+                c1,
+                r1,
+                color_stops: color_stops.to_vec(),
+                extend,
+            },
+            Brush::SweepGradient {
+                c0,
+                start_angle,
+                end_angle,
+                color_stops,
+                extend,
+            } => BrushParams::SweepGradient {
+                c0,
+                start_angle,
+                end_angle,
+                color_stops: color_stops.to_vec(),
+                extend,
+            },
+        }
+    }
+}
+
+// Needed as a mirror struct with owned ColorStops for serialization, deserialization.
+#[derive(Serialize, Deserialize, PartialEq)]
+pub enum BrushParams {
+    Solid {
+        palette_index: u16,
+        alpha: f32,
+    },
+    // Normalized to a straight line between points p0 and p1,
+    // color stops normalized to align with both points.
+    LinearGradient {
+        p0: Point<f32>,
+        p1: Point<f32>,
+        color_stops: Vec<ColorStop>,
+        extend: Extend,
+    },
+    RadialGradient {
+        c0: Point<f32>,
+        r0: f32,
+        c1: Point<f32>,
+        r1: f32,
+        color_stops: Vec<ColorStop>,
+        extend: Extend,
+    },
+    SweepGradient {
+        c0: Point<f32>,
+        start_angle: f32,
+        end_angle: f32,
+        color_stops: Vec<ColorStop>,
+        extend: Extend,
+    },
+}
+
+#[derive(Serialize, Deserialize, PartialEq)]
+enum PaintOps {
+    PushTransform {
+        transform: Transform,
+    },
+    PopTransform,
+    PushClipGlyph {
+        gid: u16,
+    },
+    PushClipRectangle {
+        x_min: f32,
+        x_max: f32,
+        y_min: f32,
+        y_max: f32,
+    },
+    PopClip,
+    FillBrush {
+        brush: BrushParams,
+    },
+    PushLayer {
+        composite_mode: u8,
+    },
+    PopLayer,
+}
+
+impl ColorPainter for JsonDumpColorPainter {
```

Comment:
Optional: we could just implement this directly for PaintDump.

Reply:
Good idea, that looks cleaner, done.